utl/clang-format

```diff
@@ -4,18 +4,31 @@
 # "./utl/clang-format fix" formats your code.
 # "./utl/clang-format check" exits with non-zero if your code isn't formatted correctly.
 # Intended to be used with clang-format 7.0.1.
+# Set environment variable GH_WORKFLOW_LOGGING to output logging that Azure pipelines will interpret as a warning.
+# Set environment variable WARNING_AS_ERROR to make the script exit with a non-zero code if issues are found.
 
 for VW_FILE in $(find vowpalwabbit -type f -name '*.cc' -o -name '*.h'); do
 
     if [ "$1" = check ]; then
-       diff $VW_FILE <(clang-format $VW_FILE);
-       if [ $? -ne 0 ]; then
-       	  echo "Run \"./utl/clang-format fix\""
-       fi
+        diff $VW_FILE <(clang-format $VW_FILE);
+        if [ $? -ne 0 ]; then
+            ISSUE_FOUND="true"
+            if [[ -v GH_WORKFLOW_LOGGING ]]; then
+                echo "::warning file=$VW_FILE:: Formatting issues found in $VW_FILE"
```

Comment:
@jackgerrits do we have access to git here? it be interesting to filter the files by the ones only being affected by this change

Reply:
might be too complicated but worthwhile to consider or discuss

Reply:
That's exactly what the workflow does :) The file you're looking at here is the general purpose "run clang-format on the whole codebase". The following is how to do it just for the PR diff:

```
git diff origin/master HEAD -U0 --no-color | clang-format-diff -r '^.*\.(cc|h)$' -p1 > clang_format_diff.txt
          if [ -s clang_format_diff.txt ]; then
            cat clang_format_diff.txt
            echo "::error:: Formatting issues found"
            exit 1
          else
            echo "No formatting issues found in the PR diff."
          fi 
```

Reply:
Nice, got misled by scrolling down

Reply:
It will be misleading until we fix the warnings :)